ext/bin/pdb-dataset

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+set -ueo pipefail
+
+usage() { echo 'Usage: PDBBOX=BOX_DIR pdb-dataset import --timeshift-to now DUMPFILE'; }
+
+misuse()
+{
+  usage 1>&2
+  exit 1
+}
+
+if test "$#" -lt 1 -o -z "$PDBBOX"; then
+  misuse
+fi
+
+test $# -gt 0 || misuse
+command="$1"
+shift 1
+
+case "$command" in
+  import)
+    timeshift=''
+    timeshiftsql=''
+    dumpfile=''
+
+    while test $# -gt 0; do
+      case "$1" in
+        --timeshift-to)
+          test $# -gt 0 || misuse
+          timeshift="$2"
+          shift 2
+          ;;
+        *)
+          if [ -z "$dumpfile" ]; then
+            dumpfile="$1"
+            shift
+          else
+            echo "Unknown argument: $1" 1>&2
+            misuse
+          fi
+      esac
+    done
+
+    if [ "$timeshift" = 'now' ]; then
+      timshiftsql="now()"
+    elif [[ $timeshift =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ ]]; then
+      timshiftsql="TIMESTAMP '${timeshift}'"
+    else
+      echo "Invalid --timeshift value: $timeshift" 1>&2
+      misuse
+    fi
+
+    echo "Checking for existing puppetdb database" 1>&2
+    db_intialized=$(psql "puppetdb" -t -c "SELECT 1 FROM information_schema.tables WHERE table_name = 'schema_migrations';")
+
+    if [ "$db_intialized" ]; then
+      echo 'puppetdb database already exists' 1>&2
+      exit 2
+    fi
+
+    echo "Restoring database from backup" 1>&2
+    pg_restore --role=postgres --no-owner --no-acl -d puppetdb "$dumpfile" || echo $?
```

Comment:
When testing this without a `dumpfile` argument the `pg_restore` command failed and then the `db_exists` check below output `'[' '' ']'` which made it pass the following check when I think it was intended to fail. This was after I created an empty puppetdb database with `createdb -E UTF8 -O puppetdb puppetdb`. I created the empty `puppetdb` to get around the error mentioned above in the `db_initialized` check.